picard/file.py

```diff
@@ -285,7 +302,7 @@ def _save_and_rename(self, old_filename, metadata):
                     except BaseException:
                         break
                     head, tail = os.path.split(head)
-            except EnvironmentError:
```

Comment:
I wonder about loops/exception handling here.
The try/except block around this second call to `_rmdir()` matches BaseException.
Dunno but doesn't look clean to me, perhaps we should use a new exception class to differentiate "skip" and "failure".

Reply:
Yes, this is a good point. Also gives us the opportunity to do better logging here, so if we get reports of failed removals of empty dirs we can actually find out why.

https://github.com/metabrainz/picard/pull/1280/commits/6f6fb28414dec7cd266809d7f03945b4b04fba26